### PR TITLE
Improve panic messages for RowSelection::and_then (#2925)

### LINKED ISSUE #2925: Fail at rowSelection `and_then` method
**Describe the bug**
```
let a = RowSelection::from(vec![
            RowSelector::select(3),
            RowSelector::skip(33),
            RowSelector::select(3),
            RowSelector::skip(33),
        ]);
        let b = RowSelection::from(vec![RowSelector::select(36), RowSelector::skip(36)]);

        assert_eq!(
            a.and_then(&b).selectors,
            vec![RowSelector::select(3), RowSelector::skip(69)]
        );
```
got 
```
---- arrow::arrow_reader::selection::tests::test_and_2 stdout ----
thread 'arrow::arrow_reader::selection::tests::test_and_2' panicked at 'called `Option::unwrap()` on a `None` value', parquet/src/arrow/arrow_reader/selection.rs:233:38
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

```
**To Reproduce**

**Expected behavior**

**Additional context**

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Closes #2925

# Rationale for this change
 

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by waitingkuo** [comment]
LGTM

File: parquet/src/arrow/arrow_reader/selection.rs [hunk 1]
```diff
@@ -223,17 +223,26 @@ impl RowSelection {
...
         let mut selectors = vec![];
         let mut first = self.selectors.iter().cloned().peekable();
         let mut second = other.selectors.iter().cloned().peekable();
 
         let mut to_skip = 0;
         while let Some(b) = second.peek_mut() {
-            let a = first.peek_mut().unwrap();
+            let a = first
+                .peek_mut()
+                .expect("selection exceeds the number of selected rows");
```
Comment (waitingkuo) [inline parquet/src/arrow/arrow_reader/selection.rs:240]: 👍

File: parquet/src/arrow/arrow_reader/selection.rs [hunk 2]
```diff
@@ -223,17 +223,26 @@ impl RowSelection {
...
 
             if b.row_count == 0 {
-                second.next().unwrap();
+                second
+                    .next()
+                    .expect("selection contains less than the number of selected rows");
```
Comment (waitingkuo) [inline parquet/src/arrow/arrow_reader/selection.rs:245]: 👍

File: parquet/src/arrow/arrow_reader/selection.rs [hunk 3]
```diff
@@ -223,17 +223,26 @@ impl RowSelection {
...
         while let Some(b) = second.peek_mut() {
-            let a = first.peek_mut().unwrap();
+            let a = first
+                .peek_mut()
+                .expect("selection exceeds the number of selected rows");
 
             if b.row_count == 0 {
-                second.next().unwrap();
+                second
+                    .next()
```
Comment (viirya) [inline parquet/src/arrow/arrow_reader/selection.rs:244]: We peek into `second` to get `b`. Why `second.next()` will be None?

**Review by viirya — APPROVED** [review]

**Pull request merged**